### ISSUE TITLE
Failed to load plugin error on line 286

### ISSUE THREAD

**Issue by openmedi** [sciunto/ikiwiki-pandoc#3]
With ikiwiki version 3.20150614 installed via pkgsrc on OS X 10.10 with Pandoc 1.15.0.6. it doesn't seem to work. First of all, the installation instructions seem incomplete, as they are not including the renaming of the page template back to it's original name, which, I assume, is what needs to happen (otherwise I get an error stating that the page.tmpl can't be found). Have I assumed wrong? Afterwards, this is the error I get:

```
$ ikiwiki --setup zettelkasten.setup 
Failed to load plugin IkiWiki::Plugin::pandoc: Can't use 'defined(@array)' (Maybe you should just omit the defined()?) at /Users/martin/.ikiwiki/IkiWiki/Plugin/pandoc.pm line 286.
Compilation failed in require at (eval 101) line 2.
BEGIN failed--compilation aborted at (eval 101) line 2.
```

P.S.: Also, uninstall instructions would be a nice thing to have.

**Comment by bk** [comment]
I just submitted pull request #4, which hopefully fixes this.

There have been many changes to pandoc in the last 2-3 years, and this module has not kept pace with these developments. Core features should be OK, however.

**Comment by sciunto** [comment]
@openmedi Can you check that master works for you?

@bk if you are familiar with pandoc's evolutions, feel free to suggest further modifications.
Originally, I made this repository because various improvements were scattered on github and I tried to gather everything. I didn't have the chance to go beyond that unfortunately.

**Comment by bk** [comment]
The most important problem is that since this module was last substantially updated, support for YAML meta blocks has been added to pandoc. The fix in PR #4 does not address this at all, so even the old type of meta block (where the first three lines start with `%`) is not handled properly if you have Pandoc 1.4 and later. Of course, a few more input formats have also been added...

I'll try to take a look at this and will probably submit a new PR in the next few days.

**Comment by bk** [comment]
I've submitted #5, which addresses the issues I mentioned and extends the module's support for some of the more advanced features of Pandoc.
